lib/emergency_banner/display.rb

```diff
@@ -0,0 +1,34 @@
+module EmergencyBanner
+  class Display
+    def enabled?
+      content && campaign_class && heading
+    end
+
+    def campaign_class
+      content[:campaign_class] if content[:campaign_class].present?
+    end
+
+    def heading
+      content[:heading] if content[:heading].present?
+    end
+
+    def short_description
+      content[:short_description] if content[:short_description].present?
+    end
+
+    def link
+      content[:link] if content[:link].present?
+    end
+
+  private
+
+    def content
+      @data ||= connect.hgetall("emergency_banner")
+      @data.symbolize_keys if @data
+    end
+
+    def connect
```

Comment:
For timeouts, I'd suggest something like 0.1. Redis in production looks like it responds to most queries within 1ms but we should give a bit of leeway for network latency.